roles/hosted_engine_setup/tasks/bootstrap_local_vm/04_engine_final_tasks.yml

```diff
@@ -48,6 +48,20 @@
     with_items:
       - cloud-init-local
       - cloud-init
+  - name: Check if keycloak is configured
+    command: otopi-config-query query -k OVESETUP_KEYCLOAK_CORE/enable -f /etc/ovirt-engine-setup.conf
+    register: keycloak_configured_out
+    ignore_errors: true
+  - name: Set admin username
+    set_fact:
+      he_admin_username: >-
+        {{ 'admin@ovirt@internalsso'
+            if keycloak_configured_out.rc == 0 and keycloak_configured_out.stdout_lines[0] == 'True'
+            else 'admin@internal'
+        }}
+  - name: Pass engine admin user to otopi
+    set_fact:
```

Comment:
This does not work, because ohe-setup's 1_otopi_json.py looks only at 'register' results. Not sure I can 'register:' the results of 'set_fact' or what this even means. Now searched a bit and could not find anything helpful. Need to do something - either hack (e.g. run command: 'echo $stuff' and register: the results) or check if I can register: set_fact or something similar.

Reply:
Maybe register the output of the `Set admin username` task to `otopi_he_admin_username`.

Reply:
Done. Still didn't check previous, so not sure if current will work either. Let's see.